src/server/routes/api/document/index.js

```diff
@@ -81,6 +81,40 @@ let createSecret = ({ secret }) => {
   );
 };
 
+let createRelatedPapers = ({ papersData, docId }) => {
+  return loadTables()
+    .then( ({ docDb, eleDb }) => {
+      let sanitize = o => {
+        delete o.intnId;
+      };
+
+      let els = [];
+      papersData.forEach( paperData => {
+        let pmid = paperData.pmid;
+        let pubmed = paperData.pubmed;
+        paperData.elements.forEach( el => {
+          els.push( _.extend( {}, el, { pmid, pubmed } ) );
+        } );
+      } );
+
+      let elsByIntn = _.groupBy( els, 'intnId' );
+
+      let elPromises = Object.keys( elsByIntn ).map( intnId => {
+        let elPapersData = elsByIntn[ intnId ];
```

Comment:
All of the interaction-level transformations seem unnecessary and end up storing redundant information for each paper. 

Given an intnId `uuid_i`, isn't the goal to simply subset the `paperData` array for those Objects whose `elements` array contains an Object with `intnId == uuid_i`?

Reply:
Possibility: Pass the `doc` from `searchRelatedPapers` on here, get the interaction Ids, then filter the `papersData` and stash the result in each interaction?

Reply:
>All of the interaction-level transformations seem unnecessary and end up storing redundant information for each paper.

I was aware of the redundancy there. Actually, we want to include the information be available as something like ``interaction.relatedPapers()`` and ``document.relatedPapers()``. For some reason I do not know I focused on making that data directly accessible under ``Element`` and ``Document`` databases as a field. However, then I realized that what we want does not require that. We can accomplish it as having a relatedPapers database table as I did in previous commits. I will probably convert back to that way and will remove the relatedPapers fields from Element and Document tables. However, it may also dependant of the ideal way of attching that information to the object as I asked somewhere above.

Reply:
Let's not worry about trying to minimise the size of the database.  Here are some things to consider:

(1) We should not be doing low-level DB operations for document or element data.  That risks creating new bugs, and it duplicates DB logic already within the classes in the model.  Use the `Document`, `Element`, etc. classes to store data for documents and elements.

(2) The API to get related papers should be as simple as something like `interaction.relatedPapers()`.  There should be no expensive processing needed to get that result.  Any expensive processing should be a part of the background/batch processing.

Reply:
Let me clarify, as is, the data stored on the interaction is not usable. It would need to be remapped into the format attached to the document.

Reply:
>Let me clarify, as is, the data stored on the interaction is not usable. It would need to be remapped into the format of papersData from which it originated

I think I had a little confusion in concepts of factiod interaction and interaction from related papers. That is why I wanted to put related papers info each of related papers interactions. I think I understand what you mean I will do like that.